doc/compiler.md

```diff
@@ -0,0 +1,40 @@
+## Miniscript Compiler
+
+This library provides a Policy compiler that converts bitcoin Policy to Miniscript which can
+be used with "compiler" feature. The compiler offers a simple way to interface with
+miniscript and often the first place beginners start to play with miniscript. There are
+however several limitations and caveats to the compiler. This document tries the compiler
+inner working as well as explains the guarantees it offers(or does not offer).
+
+### The Miniscript compiler
+
+As mentioned, the policy compiler takes in input a policy and converts it to a miniscript. The
+compiler algorithm is essentially a brute force based algorithm with some heuristics to
+reduce the search space. For each sub-policy, and for each miniscript type, satisfaction probability
+and dissatisfaction probability, the compiler tries to find a miniscript that minimizes the spending
+weight. The spending weight here refers
+
+`spending_weight = spk_cost + witness_cost`
+
+where `spk_cost` is the cost of the scriptPubKey and `witness_cost` is the expected cost of the witness
+with the given satisfaction and dissatisfaction probabilities.
+
+### Compiler guarantees/limitations
+
+If the compiler is able to output a miniscript
+
+- the miniscript it produces is a valid miniscript.
+- It also guarantees that the miniscript produce will the spendable(assuming available witness) under the standardness rules.
```

Comment:
typo: produce**d**

Also space before the `(`.

Reply:
The `d` at the end of `produce` still needs to be addded.

Reply:
My bad. I missed reading that